docs/data/horizon/api-reference/errors/error-handling.mdx

```diff
@@ -0,0 +1,214 @@
+
+### Polling and Retrying Transactions
+
+#### Async Transaction Submission
+
+Submissions using the `/transactions_async` endpoint return an immediate response back from Stellar-Core. There are different actions that clients can take based on the specific `tx_status` returned:
+
+1. `PENDING`: The submission is successful but the transaction is still waiting to be included in a ledger. You should use the [GET /transactions/:transaction_hash](/docs/data/horizon/api-reference/resources/retrieve-a-transaction) endpoint to poll for the submitted transaction and check if it makes into a ledger. Note that even though the submission was successful, it can still fail to get included in the ledger.
+2. `ERROR`: The submission did not go through due to an error in Stellar-Core. Take a look at the attached error message for more details, modify your transaction if necessary, and resubmit.
+3. `TRY_AGAIN_LATER`: This indicates that the Stellar-Core instance is currently unable to process submission of this particular transaction. Clients should wait for sometime before resubmitting the transaction. This could happen due to different reasons:
+   - There is another transaction from same source account in memory
+   - It has been rejected due to too low fee and has been resubmitted too soon
+   - The transaction has resource consumption higher than the network limits allow: clients should wait for sometime before retrying it again.
+
+<CodeExample>
+
+```js
+let server = sdk.Server("https://horizon-testnet.stellar.org");
+let contractId = "CA3D5KRYM6CB7OWQ6TWYRR3Z4T7GNZLKERYNZGGA5SOAOPIFY6YQGAXE";
+let contract = new StellarSdk.Contract(contractId);
+
+// Right now, this is just the default fee for this example.
+const fee = StellarSdk.BASE_FEE;
+
+let transaction = new StellarSdk.TransactionBuilder(account, { fee })
+  .setNetworkPassphrase(StellarSdk.Networks.TESTNET)
+  .setTimeout(30) // valid for the next 30s
+  // Add an operation to call increment() on the contract
+  .addOperation(contract.call("increment"))
+  .build();
+
+// Sign this transaction with the secret key
+// NOTE: signing is transaction is network specific. Test network transactions
+// won't work in the public network. To switch networks, use the Network object
+// as explained above (look for StellarSdk.Network).
+let sourceKeypair = StellarSdk.Keypair.fromSecret(sourceSecretKey);
+transaction.sign(sourceKeypair);
+
+server.submitAsyncTransaction(transaction).then((result) => {
+  console.log("hash:", result.hash);
+  console.log("status:", result.tx_status);
+  console.log("errorResultXdr:", result.error_result_xdr);
+});
+```
+
+</CodeExample>
```

Comment:
Can we demonstrate taking the result hash and using it to poll the transaction? I think thats the missing last step here.

Reply:
Agreed, but let's try not to get too fancy with polling so that people don't blindly copy-paste the fetching code (e.g. just add a sleep before the GET).